src/gui/toolbarMenubar/model/ToolbarColorNames.cpp

```diff
@@ -56,22 +32,51 @@ void ToolbarColorNames::saveFile(fs::path const& file) {
     g_free(data);
 }
 
+
+ToolbarColorNames::ToolbarColorNames(fs::path readFile): file(std::move(readFile)) {
+    this->config = g_key_file_new();
+    g_key_file_set_string(this->config, "info", "about", "Xournalpp custom color names");
+    initPredefinedColors();
+    load(this->file, this->config, predefinedColorNames);
+}
+
+ToolbarColorNames::~ToolbarColorNames() {
+    saveFile(file);
```

Comment:
Saving the file should be explicit in the application shutdown--don't do it here in the destructor.